### PR TITLE
Migration of cadence-lang.org to Docusaurus3

### PR DESCRIPTION
**Pull request by bthaile**
We are migrating cadence-lang.org website to Docusaurus2, becaue we want to move the Language docs from flow developers portal to cadence-lang.org.

Tasks:
- [x] Migrate the Docs content.
- [x] Fix links and layout issues.
- [ ] Get the site hosted in vercel.

created a testing vercel deployment
get current testing url below for `cadence-lang-docs` vercel url.

### CODE REVIEW

File: blog/2019-05-28-first-blog-post.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
```
Comment (turbolent) [inline blog/2019-05-28-first-blog-post.md:1]: This doesn't look right, this adds lots of sample Docusaurus example content

**Review by turbolent — CHANGES_REQUESTED** [review]

**Comment by turbolent** [comment]
Is this a WIP PR? Please turn it into a draft if that's the case

File: blog/2019-05-28-first-blog-post.md (hunk 1 above)
Comment (bthaile) [reply blog/2019-05-28-first-blog-post.md:1]: This PR is for @j1010001  as part of the effort to integrate docusaurus to cadence-lang site.

**Comment by bthaile** [comment]
> Is this a WIP PR? Please turn it into a draft if that's the case.
> 
> The title also seems off. TBH, I don't quite know what to make of this PR, maybe we can sync about it?

This PR is for @j1010001 as part of the effort to integrate docusaurus to cadence-lang site.

**Comment by turbolent** [comment]
Thank you for porting the website to Docusaurus and moving the documentation over to the website.

Overall the site looks good, well done! 👏 

Some feedback:

- Versions should just be named `1.0` and `0.42`. Currently the naming with the "Cadence" prefix leads to messages in the UI like "This is unreleased documentation for Cadence **Cadence 1.0** version."
- Syntax highlighting is missing
- The layout / typography in the documentation seems a bit off. It might be good to adjust some of it, like paddings, margins, font sizes, etc.
 Compare e.g. https://cadence-lang-docs-qld92jrev-onflow.vercel.app/docs/next/ with https://developers.flow.com/cadence/intro

**Comment by bthaile** [comment]
> * https://cadence-lang-docs-qld92jrev-onflow.vercel.app/docs/next/

In reference to the last item
> "Compare e.g. https://cadence-lang-docs-qld92jrev-onflow.vercel.app/docs/next/ with https://developers.flow.com/cadence/intro"

I was preserving the original cadence-lang style and adding docusaurus. So the style will be different than the developer docs.

**Comment by turbolent** [comment]
@bthaile [834f23e](https://github.com/onflow/cadence-lang.org/pull/12/commits/834f23e7d36600ba08de2b7958cf86c7789d4200) was fine, I should have been more specific, I'll comment in the Vercel preview

**Comment by bthaile** [comment]
> @bthaile The margins removed in [834f23e](https://github.com/onflow/cadence-lang.org/pull/12/commits/834f23e7d36600ba08de2b7958cf86c7789d4200) were fine, can you please restore them? I should have been more specific, I meant the margins in the documentation section I'll comment in the Vercel preview

There are layout conflicts with the old cadence-lang layout and docusaurus documentation layout, Example is the top nav bar, should the "Cadence" in the top left be aligned with "Learn Cadence" on the main page or should "Cadence" be aligned with side bar on the documentation page? 
Either one layout works but need to pick one.

**Comment by turbolent** [comment]
I'd probably say the left and right edge (width) of the top navigation bar should match the content. On the landing page the content is "narrower" because there is no navigation on the left and no TOC on the right, but in the documentation the content is wider because of the left and right parts.

File: src/css/custom.css [hunk 2]
```diff
@@ -0,0 +1,491 @@
...
+  max-width: 80rem;
+  margin-left: auto;
+  margin-right: auto;
+  padding-left: max(env(safe-area-inset-left), 1.5rem);
+  padding-right: max(env(safe-area-inset-right), 1.5rem);
+}
+
+.main-wrapper {
+  max-width: 80rem;
+  min-width: 85%;
```
[the remark this answers is not in the document]
Comment (bthaile) [reply src/css/custom.css]: Nice, good one. looks good

**Comment by nialexsan** [comment]
@turbolent I've fixed the width, now it's the same for all pages

**Comment by turbolent** [comment]
@nialexsan Nice, thank you!

The page is looking nice, great work and thank you everyone!

**Review by turbolent — APPROVED** [review]

**Comment by turbolent** [comment]
After merging the PR we might want to make sure to update the documentation content with the latest changes from the `docs` repo so we don't loose anything.

There are still a couple design/layout things that are broken, but we can fix them in a follow-up PR

**Comment by turbolent** [comment]
`npm i` shows:

```
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: flux@4.0.4
npm WARN Found: react@18.2.0
npm WARN node_modules/react
npm WARN   react@"^18.2.0" from the root project
npm WARN   41 more (@code-hike/mdx, @docsearch/react, @docusaurus/core, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer react@"^15.0.2 || ^16.0.0 || ^17.0.0" from flux@4.0.4
npm WARN node_modules/@microlink/react-json-view/node_modules/flux
npm WARN   flux@"~4.0.1" from @microlink/react-json-view@1.23.0
npm WARN   node_modules/@microlink/react-json-view
npm WARN
npm WARN Conflicting peer dependency: react@17.0.2
npm WARN node_modules/react
npm WARN   peer react@"^15.0.2 || ^16.0.0 || ^17.0.0" from flux@4.0.4
npm WARN   node_modules/@microlink/react-json-view/node_modules/flux
npm WARN     flux@"~4.0.1" from @microlink/react-json-view@1.23.0
npm WARN     node_modules/@microlink/react-json-view
npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
```

Is there something that needs to be resolved?

**Comment by nialexsan** [comment]
> `npm i` shows:
> 
> ```
> npm WARN ERESOLVE overriding peer dependency
> npm WARN While resolving: flux@4.0.4
> npm WARN Found: react@18.2.0
> npm WARN node_modules/react
> npm WARN   react@"^18.2.0" from the root project
> npm WARN   41 more (@code-hike/mdx, @docsearch/react, @docusaurus/core, ...)
> npm WARN
> npm WARN Could not resolve dependency:
> npm WARN peer react@"^15.0.2 || ^16.0.0 || ^17.0.0" from flux@4.0.4
> npm WARN node_modules/@microlink/react-json-view/node_modules/flux
> npm WARN   flux@"~4.0.1" from @microlink/react-json-view@1.23.0
> npm WARN   node_modules/@microlink/react-json-view
> npm WARN
> npm WARN Conflicting peer dependency: react@17.0.2
> npm WARN node_modules/react
> npm WARN   peer react@"^15.0.2 || ^16.0.0 || ^17.0.0" from flux@4.0.4
> npm WARN   node_modules/@microlink/react-json-view/node_modules/flux
> npm WARN     flux@"~4.0.1" from @microlink/react-json-view@1.23.0
> npm WARN     node_modules/@microlink/react-json-view
> npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
> ```
> 
> Is there something that needs to be resolved?

no, that's just a warning from one of the debug packages, nothing to worry about

**Comment by bthaile** [comment]
New deployment is up and running.  https://cadence-lang.org/

**Pull request merged**